otx/cli/builder/builder.py

```diff
@@ -155,33 +156,34 @@ def build_task_config(
         """
 
         # Create OTX-workspace
-        # Check whether the workspace is existed or not
-        workspace_path.mkdir(exist_ok=False)
+        workspace_path.mkdir(parents=True, exist_ok=False)
 
         # Load & Save Model Template
-        otx_registry = OTXRegistry(otx_root).filter(task_type=task_type)
-        if model_type:
-            template_lst = [temp for temp in otx_registry.templates if temp.name.lower() == model_type.lower()]
-            if len(template_lst) == 0:
-                raise ValueError(
-                    f"[otx build] {model_type} is not a type supported by OTX {task_type}."
-                    f"\n[otx build] Please refer to 'otx find --template --task_type {task_type}'"
-                )
-            template = template_lst[0]
-        else:
-            template = otx_registry.get(DEFAULT_MODEL_TEMPLATE_ID[task_type.upper()])
-        template_dir = Path(template.model_template_path).parent
+        if template is None:
+            otx_registry = OTXRegistry(otx_root).filter(task_type=task_type)
+            if model_type:
+                template_lst = [temp for temp in otx_registry.templates if temp.name.lower() == model_type.lower()]
+                if len(template_lst) == 0:
+                    raise ValueError(
+                        f"[otx build] {model_type} is not a type supported by OTX {task_type}."
+                        f"\n[otx build] Please refer to 'otx find --template --task_type {task_type}'"
+                    )
+                template = template_lst[0]
+            else:
+                template = otx_registry.get(DEFAULT_MODEL_TEMPLATE_ID[task_type.upper()])
+
+        template_dir: Path = Path(template.model_template_path).parent
 
         # Copy task base configuration file
-        task_configuration_path = template_dir / template.hyper_parameters.base_path
+        task_configuration_path = template_dir / str(template.hyper_parameters.base_path)
         shutil.copyfile(task_configuration_path, str(workspace_path / "configuration.yaml"))
         # Load Model Template
         template_config = MPAConfig.fromfile(template.model_template_path)
         template_config.hyper_parameters.base_path = "./configuration.yaml"
 
         # Configuration of Train Type value
         train_type_rel_path = ""
-        if train_type != "incremental":
+        if train_type is not None and train_type != "incremental":
```

Comment:
There seems to be no case where None is entered in train_type, but isn't it?

Reply:
It'll be removed ! according to @eunwoosh 's opinion, I revmoed `Optional` for `train_type`. So, I `train_type is not None` can be removed